test/integration_test/kubevirt_test.go

```diff
@@ -246,3 +258,65 @@ func createDatadiskTemplates(t *testing.T) error {
 	}
 	return nil
 }
+
+func kubevirtVMScaledDeployAndValidate(t *testing.T, instanceID, appKey string, scale int, allCtx *([]*scheduler.Context)) {
+	var wg sync.WaitGroup
+	logrus.Infof("Deploying %d VMs and validating them.", scale)
+	for i := 1; i <= scale; i++ {
+		logrus.Infof("Deploying VM: %d", i)
+		wg.Add(1)
+		ctxs := kubevirtVMsDeployAndValidate(
+			t,
+			instanceID+"-"+fmt.Sprintf("%d", i),
+			[]string{appKey},
+			&wg,
+		)
+		logrus.Infof("Validated VM: %d", i)
+		*allCtx = append(*allCtx, ctxs...)
+	}
+	wg.Wait()
+}
+
+func createClusterroleDataVolume(clusterroleName string) error {
+	clusterRoleDatavolume := &rbacv1.ClusterRole{
+		ObjectMeta: meta.ObjectMeta{
+			Name: clusterroleName,
+		},
+		Rules: []rbacv1.PolicyRule{{
+			Verbs:     []string{"*"},
+			APIGroups: []string{"*"},
+			Resources: []string{"*"},
+		},
+		},
+	}
+	_, err := rbac.Instance().CreateClusterRole(clusterRoleDatavolume)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+// creates/updates clusterrolebindings by adding default ServiceAccount in the given namespace
+func createClusterroleBindingDatavolume(clusterRoleName, namespace string) error {
+	newSubject := rbacv1.Subject{Kind: "ServiceAccount", Name: "default", Namespace: namespace}
+
+	// append subjects if clusterrolebinding already exists else create new cluster role binding
+	cB, err := rbac.Instance().GetClusterRoleBinding(dataVolumeClusterroleBinding)
+	if err == nil && cB != nil {
+		cB.Subjects = append(cB.Subjects, newSubject)
+		_, err = rbac.Instance().UpdateClusterRoleBinding(cB)
+	} else {
```

Comment:
We should do this only if err != NotFound?

Reply:
If we are here then it means we could not get the cluster role binding so we create one here, in line:313 we are already checking for the `AlreadyExists` error so don't think  `if err != NotFound` is required here.